### PR TITLE
Added useragent to PingPongMessage

### LINKED ISSUE #1852: Add node useragent to pingpong messages
Add a "user agent" string to `PingPongMessage` in tari_p2p liveness and log the user agent string.

E.g. `tari/basenode/0.2.3`

This should help determine if a node is talking to an old node.

### PR DESCRIPTION
**Pull request by StriderDM**
## Description
Added useragent to PingPongMessage as well as logging of the useragent

## Motivation and Context
Closes #1852

## How Has This Been Tested?
cargo test --all

## Types of changes
* [ ] Bug fix (non-breaking change which fixes an issue)
* [x] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [ ] New Tests
* [ ] Documentation

## Checklist:
* [x] I'm merging against the `development` branch.
* [x] I ran `cargo-fmt --all` before pushing.
* [x] I have squashed my commits into a single commit.
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [ ] I have added tests to cover my changes.

### CODE REVIEW

File: base_layer/p2p/src/proto/tari.p2p.liveness.rs [hunk 1]
```diff
@@ -10,6 +10,9 @@ pub struct PingPongMessage {
     /// Metadata attached to the message. The int32 key SHOULD always be one of the keys in `MetadataKey`.
     #[prost(map = "int32, bytes", tag = "3")]
     pub metadata: ::std::collections::HashMap<i32, std::vec::Vec<u8>>,
+    /// Indicates the application version from which the message was sent
```
Comment (sdbondi) [inline base_layer/p2p/src/proto/tari.p2p.liveness.rs:13]: You'll need to update the liveness.proto file which will generate this struct :)

Comment (StriderDM) [reply base_layer/p2p/src/proto/tari.p2p.liveness.rs:13]: Thanks.

File: base_layer/p2p/src/services/liveness/service.rs [hunk 2]
```diff
@@ -178,11 +178,17 @@ where
         } = msg;
         let node_id = source_peer.node_id;
         let public_key = source_peer.public_key;
+        let msg_useragent = ping_pong_msg.useragent.clone();
```
Comment (sdbondi) [inline base_layer/p2p/src/services/liveness/service.rs]: Save a clone: Remove this and use the message directly in logs. In a few other places too - I don't think (could be wrong) that you'd need to clone the incoming user agent string at all

File: base_layer/p2p/src/services/liveness/service.rs [hunk 3]
```diff
@@ -178,11 +178,17 @@ where
...
         match ping_pong_msg.kind().ok_or_else(|| LivenessError::InvalidPingPongType)? {
             PingPong::Ping => {
                 self.state.inc_pings_received();
-                trace!(target: LOG_TARGET, "Received ping from peer '{}'", node_id.short_str());
+                trace!(
+                    target: LOG_TARGET,
+                    "Received ping from peer '{}' with useragent {}",
+                    node_id.short_str(),
+                    msg_useragent
+                );
```
Comment (sdbondi) [inline base_layer/p2p/src/services/liveness/service.rs]:
```suggestion
                trace!(
                    target: LOG_TARGET,
                    "Received ping from peer '{}' with useragent {}",
                    node_id.short_str(),
                    ping_pong_msg.useragent
                );
```

**Review by sdbondi — COMMENTED** [review]
Looks good - just some clone nits

File: base_layer/p2p/src/services/liveness/service.rs (hunk 2 above)
Comment (StriderDM) [reply base_layer/p2p/src/services/liveness/service.rs]: You are correct, updated.

File: base_layer/wallet/tests/transaction_service/service.rs [hunk 4]
```diff
@@ -182,6 +182,7 @@ pub fn setup_transaction_service<T: TransactionBackend + Clone + 'static>(
                 refresh_neighbours_interval: Default::default(),
                 refresh_random_pool_interval: Default::default(),
                 random_peer_selection_ratio: 0.0,
+                useragent: format!("{}{}", "tari\\", env!("CARGO_PKG_VERSION")),
```
Comment (sdbondi) [inline base_layer/wallet/tests/transaction_service/service.rs]: Just a test so might as well be an empty string

**Review by sdbondi — COMMENTED** [review]
I just realised that the user agent string needs to be hooked up in tari_base_node builder.rs

File: applications/tari_base_node/src/builder.rs [hunk 5]
```diff
@@ -1160,6 +1161,7 @@ async fn register_wallet_services(
             LivenessConfig{
                 auto_ping_interval: None,
                 enable_auto_join: true,
+                useragent: format!("{}{}","tari\\basewallet\\",env!("CARGO_PKG_VERSION")),
```
Comment (sdbondi) [inline applications/tari_base_node/src/builder.rs]:
```suggestion
                useragent: format!("{}{}","tari\\wallet\\",env!("CARGO_PKG_VERSION")),
```

Comment (StriderDM) [reply applications/tari_base_node/src/builder.rs]: Done

File: base_layer/wallet/tests/transaction_service/service.rs (hunk 4 above)
Comment (StriderDM) [reply base_layer/wallet/tests/transaction_service/service.rs]: Done

File: base_layer/wallet/src/wallet.rs [hunk 6]
```diff
@@ -142,6 +142,7 @@ where
                 LivenessConfig {
                     auto_ping_interval: Some(Duration::from_secs(30)),
                     enable_auto_join: true,
+                    useragent: format!("{}{}", "tari\\wallet\\", env!("CARGO_PKG_VERSION")),
```
Comment (sdbondi) [inline base_layer/wallet/src/wallet.rs]:
```suggestion
                    useragent: format!("tari\\wallet\\{}", env!("CARGO_PKG_VERSION")),
```

Comment (StriderDM) [reply base_layer/wallet/src/wallet.rs]: Done.

**Review by sdbondi — APPROVED** [review]
LGTM

**Pull request merged**